### PR TITLE
fix default incorrectly overriding FrontendAddress

### PR DESCRIPTION
**Pull request by jakubgs**
The issue introduced in https://github.com/cortexproject/cortex/commit/eba13a81e73d9835a9adaf29ce6d8cd0d385c44f that causes `Worker.FrontendAddress` to be overridden by the default even when set by user.

Resolves: https://github.com/cortexproject/cortex/issues/3644

Ping: @pstibrany

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
LGTM, thanks!

**Comment by pstibrany** [comment]
This fixes a bug. Can you please also add a CHANGELOG entry?

**Comment by jakubgs** [comment]
>Can you please also add a CHANGELOG entry?

Ah, good point. There you go.

File: CHANGELOG.md
Comment (pstibrany) [inline CHANGELOG.md:31]:
```suggestion
* [BUGFIX] Querier: fix default value incorrectly overriding `-querier.frontend-address` in single-binary mode.  #3650
```

Comment (jakubgs) [reply CHANGELOG.md:31]: I guess I was confused by the file naming(`pkg/querier/worker/worker.go`) and config section name `frontend_worker`.

Comment (pstibrany) [reply CHANGELOG.md]: No worries, it *is* confusing. Querier is the top-level Cortex component, which internally runs "frontend worker" that connects to frontend component. Thanks for your changes!

**Comment by jakubgs** [comment]
Is there any timeline for `1.6.1` release that would include this? Or should I just roll back to `1.5.0` for the time being?

**Comment by pstibrany** [comment]
> Is there any timeline for `1.6.1` release that would include this? Or should I just roll back to `1.5.0` for the time being?

We rarely do patch-point releases. Perhaps you can use master instead, once this is merged? (PR needs another approval first) We don't have binaries, but there are docker images built from master.

**Comment by jakubgs** [comment]
>We rarely do patch-point releases.

And ignoring configuration for a core component is not enough to make a patch release?

Ah well, then I guess I'll just build master. I assume `make dist` is what I want.

**Review by khaines — APPROVED** [review]
LGTM!

**Pull request merged**
